ansible_collections/arista/avd/plugins/plugin_utils/cv_client/client/swg.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (c) 2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+from datetime import datetime
+from typing import TYPE_CHECKING, Literal
+
+from ..api.arista.swg.v1 import (
+    EndpointConfig,
+    EndpointConfigServiceStub,
+    EndpointConfigSetRequest,
+    EndpointStatus,
+    EndpointStatusServiceStub,
+    EndpointStatusStreamRequest,
+    SwgKey,
+)
+from .exceptions import get_cv_client_exception
+
+if TYPE_CHECKING:
+    from .cv_client import CVClient
+
+
+class SwgMixin:
+    """
+    Only to be used as mixin on CVClient class.
+    """
+
+    swg_api_version: Literal["v1"] = "v1"
+
+    async def set_swg_device(
+        self: CVClient,
+        device_id: str,
+        service: Literal["zscaler"],
+        location: str,
+        time: datetime | None = None,
+        timeout: float = 10.0,
+    ) -> EndpointConfig:
+        """
+        Get SWG Endpoints using arista.swg.v1.EndpointStatusService.GetOne API
+
+        Parameters:
+            device_id: Unique identifier of the Device - typically serial number.
+            service: SWG service
+            location: Device location/address. CloudVision will resolve a location from this address.
+            time: Timestamp from which the information is fetched. `now()` if not set.
+            timeout: Timeout in seconds.
+
+        Returns:
+            EndpointConfig including any server-generated values.
+        """
+        request = EndpointConfigSetRequest(
+            key=SwgKey(device_id=device_id, service=service),
+            time=time,
+        )
+        client = EndpointConfigServiceStub(self._channel)
+
+        try:
+            response = await client.set(request, metadata=self._metadata, timeout=timeout)
+            return response.value
+
+        except Exception as e:
+            raise get_cv_client_exception(e, f"Device ID '{device_id}', service '{service}', location '{location}'") or e
+
+    async def wait_for_swg_endpoint_status(
+        self: CVClient,
+        device_id: str,
+        service: Literal["zscaler"],
+        time: datetime | None = None,
+        timeout: float = 10.0,
+    ) -> EndpointStatus:
+        """
+        Subscribe and wait for one SWG Endpoint using arista.swg.v1.EndpointStatusService.Subscribe API
+
+        Parameters:
+            device_id: Unique identifier of the Device - typically serial number.
+            time: Timestamp from which the information is fetched. `now()` if not set.
+            timeout: Timeout in seconds.
+
+        Returns:
+            EndpointStatus object matching the device_id
+        """
+        request = EndpointStatusStreamRequest(
+            partial_eq_filter=[
+                EndpointStatus(
+                    key=SwgKey(device_id=device_id, service=service),
+                ),
+            ],
+            time=time,
```

Comment:
this also seems incorrect, AFAIK, this field is supposed to be timebounds, not time.